lib/dict.c

```diff
@@ -42,7 +42,8 @@ int rc_read_dictionary (rc_handle *rh, char const *filename)
 	char            attrstr[AUTH_ID_LEN];
 	char            typestr[AUTH_ID_LEN];
 	char            optstr[AUTH_ID_LEN];
-	char            *cp, *ifilename;
+	char            ifilename[AUTH_ID_LEN] = {0};
```

Comment:
not your fault but `AUTH_ID_LEN` has nothing to do with limits of filenames and is in fact very small (64). What about using `PATH_MAX`?